service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterMapper.java

```diff
@@ -30,9 +47,49 @@ public String mapEncounterToEhrComposition(Encounter encounter) {
             .effectiveTime(StatementTimeMappingUtils.prepareEffectiveTimeForEncounter(encounter))
             .availabilityTime(StatementTimeMappingUtils.prepareAvailabilityTimeForEncounter(encounter))
             .status(COMPLETE_CODE)
-            .components(components);
+            .components(components)
+            .code(buildCode(encounter));
 
         return TemplateUtils.fillTemplate(ENCOUNTER_STATEMENT_TO_EHR_COMPOSITION_TEMPLATE,
             encounterStatementTemplateParameters.build());
     }
+
+    private String buildCode(Encounter encounter) {
+        var type = encounter.getType()
+            .stream()
+            .findFirst()
+            .orElseThrow(() -> new EhrMapperException("Could not map Encounter type"));
+
+        var coding = type.getCoding()
+            .stream()
+            .filter(this::isSnomedAndWithinEhrCompositionVocabularyCodes)
+            .findFirst();
+
+        if (coding.isPresent()) {
+            return String.format(EHR_COMPOSITION_SNOMED_CODE, coding.get().getCode(), coding.get().getDisplay(), buildCodeText(type));
+        } else {
+            return String.format(EHR_COMPOSITION_OTHER_REPORT_CODE, buildCodeText(type));
+        }
+    }
+
+    private boolean isSnomedAndWithinEhrCompositionVocabularyCodes(Coding coding) {
+        return coding.hasSystem() && coding.getSystem().equals(SNOMED_SYSTEM)
+            && Arrays.asList(EHR_COMPOSITION_NAME_VOCABULARY_CODES).contains(coding.getCode());
+    }
+
+    private String buildCodeText(CodeableConcept codeableConcept) {
+        if (codeableConcept.hasText()) {
+            return String.format(EHR_COMPOSITION_CODE_TEXT, codeableConcept.getText());
+        } else {
+            return StringUtils.EMPTY;
+        }
+    }
+
+    @SneakyThrows
+    private static String[] getEhrCompositionNameVocabularyCodes() {
+        InputStream inputStream = EncounterMapper.class.getClassLoader().getResourceAsStream("ehr_composition_name_vocabulary_codes.txt");
```

Comment:
Possible improvement: Put these into an unmodifiable Set (HashSet).
Lookups will be must faster. Currently, for every encounter, a new List is created from the array and worse case  contains has to iterate the entire list.
Also the array, although final, could be modified at runtime.